### PR TITLE
Use java toolchains to setup JDK 8 path

### PR DESCRIPTION
**Pull request by ilya-g**
Also remove commented out JVM 6 target.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -362,8 +362,10 @@ git submodule init
...
 ```
 
-The path to JDK 8 must be specified either with the environment variable `JDK_8` or 
-with the gradle property `JDK_8`. For local builds, you can use a later version of JDK if you don't have that 
-version installed.
+The project requires JDK 8 to build classes and to run tests. 
+Gradle will try to find it among the installed JDKs or [provision](https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning) it automatically if it couldn't be found.
+The path to JDK 8 can be additionally specified with the environment variable `JDK_8`. 
+<!-- For local builds, you can use a later version of JDK if you don't have that 
+version installed. -->
```
Comment (ilya-g) [inline README.md:369]: This is no longer possible, but I plan to add a gradle property for overriding the main Java toolchain version used in the build later.

Comment (ilya-g) [reply README.md]: Now can be overridden with the property `java.mainToolchainVersion`, provided that that toolchain is detected or provisioned by Gradle.

**Review by dkhalanskyjb — APPROVED** [review]
Checked that the `jvmTest` for both subprojects runs on Linux and MacOS, whether or not the environment variables are set. Also, on Linux, the project builds in general.

For some reason, on the Mac laptop, even though it's not ARM-based, the build attempts to also run tests for the arm64 targets, which fails the build for me, but the same also happens on `master`.

File: README.md [hunk 2]
```diff
@@ -362,8 +362,11 @@ git submodule init
...
 ```
 
-The path to JDK 8 must be specified either with the environment variable `JDK_8` or 
-with the gradle property `JDK_8`. For local builds, you can use a later version of JDK if you don't have that 
-version installed.
+The project requires JDK 8 to build classes and to run tests. 
+Gradle will try to find it among the installed JDKs or [provision](https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning) it automatically if it couldn't be found.
+The path to JDK 8 can be additionally specified with the environment variable `JDK_8`. 
+
+> For local builds, you can use a later version of JDK if you don't have that
```
[the remark this answers is not in the document]
Comment (ilya-g) [reply README.md:369]: This is done to render this note in the blockquote format. Do you think it's not worth to draw attention to this note?

[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply README.md:369]: Ah, ok. I just expected to see a quote in a blockquote, and *italics* for emphasis, for example. However, now I see that we already use block quotes for emphasis.

That said, now that you ask, I do not think this should be emphasized. Looking at https://github.com/Kotlin/kotlinx-datetime/tree/toolchains#building, I think the section is nice and compact, difficult to get lost in.

[the remark this answers is not in the document]
Comment (ilya-g) [reply README.md]: ok, removed emphasis.

**Pull request merged**